Generator.Application/Handlers/GetRandomPicturesQueryHandler.cs

```diff
@@ -7,28 +7,36 @@
 using Generator.Application.Dtos;
 using Generator.Application.Persistence;
 using Generator.Application.Queries;
+using Generator.Application.Security;
 using MediatR;
 
 namespace Generator.Application.Handlers
 {
-    public class GetRandomPicturesQueryHandler : IRequestHandler<GetRandomPicturesQuery, List<PictureDto>>
+    public class GetRandomPicturesQueryHandler : IRequestHandler<GetRandomPicturesQuery, RandomPicturesResponseDto>
     {
         private readonly GeneratorContext context;
         private readonly IMapper mapper;
+        private readonly ISecurityTokenService securityTokenService;
 
-        public GetRandomPicturesQueryHandler(GeneratorContext context, IMapper mapper)
+        public GetRandomPicturesQueryHandler(GeneratorContext context, IMapper mapper, ISecurityTokenService securityTokenService)
         {
             this.context = context;
             this.mapper = mapper;
+            this.securityTokenService = securityTokenService;
         }
 
-        public Task<List<PictureDto>> Handle(GetRandomPicturesQuery query, CancellationToken token)
+        public async Task<RandomPicturesResponseDto> Handle(GetRandomPicturesQuery query, CancellationToken cancellationToken)
         {
             var pictures = context.Pictures.Where(e =>
                     context.Pictures.Select(x => x.Id).OrderBy(r => Guid.NewGuid()).Take(2).Contains(e.Id)).ToList();
             var pictureDtoA = mapper.Map<PictureDto>(pictures[0]);
             var pictureDtoB = mapper.Map<PictureDto>(pictures[1]);
-            return Task.FromResult(new List<PictureDto>() { pictureDtoA, pictureDtoB });
+
+            var token = await securityTokenService.SaveDataWithTokenAsync(
+                new List<string> { pictureDtoA.Id.ToString(), pictureDtoB.Id.ToString() },
+                query.UserId);
+
+            return await Task.FromResult(new RandomPicturesResponseDto(pictureDtoA, pictureDtoB, token));
```

Comment:
CS4016:  Since this is an async method, the return expression must be of type 'RandomPicturesResponseDto' rather than 'Task<RandomPicturesResponseDto>'

Reply:
no to nie rób async?

Reply:
Wcześniej takie coś mogło być możliwe, ale po zmienieniu Single na SingleAync coś innego awaituje w tej metodzie, więc pozostanie ona async